src/Stryker.Core/Stryker.Core/Mutants/NodeOrchestratorBase.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Extensions.Logging;
+using Stryker.Core.Logging;
+using Stryker.Core.Mutators;
+
+namespace Stryker.Core.Mutants.CsharpNodeOrchestrators;
+
+internal abstract class NodeOrchestratorBase
+{
+    protected static readonly Regex _pattern = new("^\\s*\\/\\/\\s*Stryker", RegexOptions.Compiled | RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(20));
+    protected static readonly Regex _parser = new("^\\s*\\/\\/\\s*Stryker\\s*(disable|restore)\\s*(once|)\\s*([^:]*)\\s*:?(.*)$", RegexOptions.Compiled | RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(20));
+    protected static readonly ILogger _logger = ApplicationLogging.LoggerFactory.CreateLogger<NodeOrchestratorBase>();
+
+    public static MutationContext ParseStrykerComment(MutationContext context, Match match, SyntaxNode node)
+    {
+        const int ModeGroup = 1;
+        const int OnceGroup = 2;
+        const int MutatorsGroup = 3;
+        const int CommentGroup = 4;
+
+        // get the ignore comment
+        var comment = match.Groups[CommentGroup].Value.Trim();
```

Comment:
Out of curiosity: what does this line get? If it gets the actual ignore comment, shouldn't the next if statement be negated? Or does a match always mean that we're dealing with an ignore comment?

Reply:
there is a lot of negation in this sentence (and in the subject at hands), so I cannot answer you beyond the first question.
This line get the reason why the subsequent mutation(s) is/are ignored, that is the user provided an explicit reason, such as `/// Sryker disable once String: Can't test this mutation`, the _ignore comment_ being `Can't Test this mutation`. Otherwise, we provide a generic one, so that at least users are made aware this was a Stryker comment that ignored this line and not some command line option

Reply:
the method `ParseStrykerComments` is supposed to be called on proper Stryker comments